quic/s2n-quic/src/provider/dc/mtu_confirm.rs

```diff
@@ -38,39 +53,51 @@ impl MtuConfirmComplete {
 }
 
 pub struct MtuConfirmContext {
-    sender: watch::Sender<State>,
+    sender: watch::Sender<MtuProbingState>,
+    peer_will_send_completion: bool,
 }
 
 impl Default for MtuConfirmContext {
     fn default() -> Self {
-        let (sender, _receiver) = watch::channel(State::default());
-        Self { sender }
+        let (sender, _receiver) = watch::channel(MtuProbingState::default());
+        Self {
+            sender,
+            // Default to true (optimistic - assume peer will send)
+            peer_will_send_completion: true,
+        }
     }
 }
 
 impl MtuConfirmContext {
-    /// Updates the state on the context
-    fn update(&mut self, state: State) {
-        self.sender.send_replace(state);
+    /// Updates the state and checks if both local and remote are complete
+    fn update_and_check(&mut self, updater: impl FnOnce(&mut MtuProbingState)) {
+        self.sender.send_modify(|state| {
+            updater(state);
+        });
     }
 }
 
 impl Drop for MtuConfirmContext {
     // make sure the application is notified that we're closing the connection
     fn drop(&mut self) {
         self.sender.send_modify(|state| {
-            if matches!(state, State::Waiting) {
-                *state = State::Ready
-            }
+            // Force ready state on connection close
+            state.local_complete = true;
+            state.remote_complete = true;
```

Comment:
I don't think I would call these fields "complete" if we set them to true on connection close. Maybe local_ready and remote_ready instead?